samples/posix/gettimeofday/src/main.c

```diff
@@ -11,6 +11,19 @@
 #include <time.h>
 #include <unistd.h>
 
+/*
+ * Make sure tv_sec is not 32 bits as that would leave
+ * applications vulnerable to 2038 issues. We have to skip
+ * this when building for glibc on 32-bit x86 as that
+ * still uses a 32-bit time_t
+ */
+#if defined(__GLIBC__) && defined(__i386__)
```

Comment:
Don't think you can assume that because glibc does allow being built with 64-bit timestamps for a 32-bit OS: https://www.mail-archive.com/bug-gnulib@gnu.org/msg41057.html

Reply:
Yeah, I'm not entirely sure what the right check is here; all I really want to avoid is causing a failure when building the test against glibc. The message linked above indicates that 64-bit values are opt-in on 32-bit systems when applications set __TIME_BITS=64? So maybe this check is OK at least in our environments where that value is not getting set?

Reply:
It is at a system level as far as I'm aware, see https://musl.libc.org/time64.html

Reply:
This check is only designed to ensure that libraries used on actual target systems are using 64 bits for time_t; what happens when Zephyr is run on a POSIX system for emulation and testing isn't something Zephyr can completely control. It *could* enable `__TIME_BITS=64` (and `__FILE_OFFSET_BITS=64`), and then this check could be removed. Actually, that's probably the best solution -- then we could always include the check making sure time_t is large enough. And maybe just do a runtime check instead of compile time as that would be less fussy.

Reply:
Yeah, that's a lot nicer. Now there's no need to skip checks on 32-bit POSIX systems -- those will be using 64-bit time_t and off_t just like all other Zephyr architectures.